doc/source/reference/arrays.classes.rst

```diff
@@ -43,6 +43,10 @@ NumPy provides several hooks that classes can customize:
 
    .. versionadded:: 1.13
 
+   .. note:: The API is `provisional
+             <https://docs.python.org/3/glossary.html#term-provisional-api>`_,
```

Comment:
Hmm, this doesn't render correctly. Maybe the link cannot be split across lines, or maybe github has a problem. We can allow long lines for links if needed.